wdl-lint/src/rules/command_shellcheck.rs

```diff
@@ -0,0 +1,357 @@
+//! A lint rule for checking mixed indentation in command text.
+use std::collections::HashMap;
+use std::collections::HashSet;
+use std::io::Write;
+use std::process;
+use std::process::Stdio;
+
+use anyhow::Context;
+use anyhow::Result;
+use anyhow::bail;
+use serde::Deserialize;
+use serde_json;
+
+use wdl_ast::AstNode;
+use wdl_ast::AstToken;
+use wdl_ast::Diagnostic;
+use wdl_ast::Diagnostics;
+use wdl_ast::Document;
+use wdl_ast::Span;
+use wdl_ast::SupportedVersion;
+use wdl_ast::SyntaxElement;
+use wdl_ast::SyntaxKind;
+use wdl_ast::ToSpan;
+use wdl_ast::VisitReason;
+use wdl_ast::Visitor;
+use wdl_ast::support;
+use wdl_ast::v1::CommandPart;
+use wdl_ast::v1::CommandSection;
+use wdl_ast::v1::Placeholder;
+use wdl_ast::v1::StrippedCommandPart;
+use wdl_ast::v1::TaskDefinition;
+
+use crate::Rule;
+use crate::Tag;
+use crate::TagSet;
+use crate::util::{count_leading_whitespace, lines_with_offset, program_exists};
+
+/// The shellcheck executable
+const SHELLCHECK_BIN: &str = "shellcheck";
+
+/// shellcheck lints that we want to suppress
+const SHELLCHECK_SUPPRESS: &[&str] = &[
```

Comment:
This is a copy-paste of miniwdl's list. I think we can remove most of these since most were added to work around their use of dummy literals, which we're not (currently) doing.

Reply:
Yeah I think this is good for now. It would be nice to make these configurable in the future. Perhaps you could add a `TODO` comment above it and/or an issue when this is merged?

Reply:
Not perfect, but users can add `#shellcheck disable=` directives right into the comment of a command section. Or use `SHELLCHECK_OPTS` to globally disable.